Build failure on windows 10 : missing execinfo.h

probably related to @johnny-bit  changes.

Clean build directory
```
F:/msys64/home/chill/darktable/src/control/signal.c:24:10: fatal error: execinfo.h: No such file or directory
   24 | #include <execinfo.h>
      |          ^~~~~~~~~~~~

```


shell script used 

```
#!/bin/bash
cd ~
#rm -R ./darktable/
git clone git://github.com/darktable-org/darktable.git

cd darktable
git pull
#git checkout 580bf49
git submodule init
git submodule update
mkdir build

cd build
cmake -G "MSYS Makefiles" -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_INSTALL_PREFIX=./debugReleaseWithDebugDT ../.
cmake --build .
cmake --build . --target install
cmake --build . --target package

```

Did not fail with missing header file but did fail later on. A snippet of output from end when it failed.

```
....
F:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/lib_darktable.dir/objects.a(styles.c.obj):F:/msys64/home/chill/darktable/src/lua/styles.c:215: more undefined references to `dt_print' follow
collect2.exe: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/lib_darktable.dir/build.make:2812: src/libdarktable.dll] Error 1
make[1]: *** [CMakeFiles/Makefile2:1321: src/CMakeFiles/lib_darktable.dir/all] Error 2
make: *** [Makefile:152: all] Error 2

```

--- a/ConfigureChecks.cmake
+++ b/ConfigureChecks.cmake
@@ -10,6 +10,8 @@ if (HAVE_CPUID_H)
     check_symbol_exists(__get_cpuid "cpuid.h" HAVE___GET_CPUID)
 endif()
 
+check_include_file(execinfo.h HAVE_EXECINFO_H)
+
 if (OpenMP_FOUND)
 
 set(CMAKE_REQUIRED_FLAGS ${OpenMP_C_FLAGS})

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -778,8 +778,10 @@ else(USE_GAME)
 endif(USE_GAME)
 
 if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-  add_definitions("-DDT_HAVE_SIGNAL_TRACE")
-  message(WARNING "Enabling signal tracing.\nrunning `-d signal --d-signal-act print-trace` will print backlogs.\nthis is solely for developers!")
+  if(HAVE_EXECINFO_H)
+    add_definitions("-DDT_HAVE_SIGNAL_TRACE")
+    message(STATUS "Signal debug: print-trace possible")
+  endif(HAVE_EXECINFO_H)
 endif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
 
 #

